core/network/src/discovery.rs

```diff
@@ -94,21 +97,30 @@ impl<TSubstream> DiscoveryBehaviour<TSubstream> {
 
 	/// Get a record from the DHT.
 	pub fn get_value(&mut self, key: &Multihash) {
-		self.kademlia.get_value(key, NonZeroU8::new(10)
-								.expect("Casting 10 to NonZeroU8 should succeed; qed"));
+		self.kademlia.get_record(key, Quorum::N(
+			NonZeroUsize::new(10).expect("10 > 0; qed")));
 	}
 
 	/// Put a record into the DHT.
 	pub fn put_value(&mut self, key: Multihash, value: Vec<u8>) {
```

Comment:
I think that the name of the api here also has to change for convenience

Reply:
Yes, but I'd like to leave that for a follow-up PR, to keep this once focused on minimal changes necessary for compatibility with the next libp2p release.